libs/community/langchain_community/vectorstores/chroma.py

```diff
@@ -33,21 +33,34 @@
 DEFAULT_K = 4  # Number of Documents to return.
 
 
-def _results_to_docs(results: Any) -> List[Document]:
-    return [doc for doc, _ in _results_to_docs_and_scores(results)]
-
-
-def _results_to_docs_and_scores(results: Any) -> List[Tuple[Document, float]]:
-    return [
-        # TODO: Chroma can do batch querying,
-        # we shouldn't hard code to the 1st result
-        (Document(page_content=result[0], metadata=result[1] or {}), result[2])
-        for result in zip(
-            results["documents"][0],
-            results["metadatas"][0],
-            results["distances"][0],
-        )
-    ]
+def _results_to_docs(
+    results: Any, include_id: Optional[bool] = False
+) -> List[Document]:
+    return [doc for doc, _ in _results_to_docs_and_scores(results, include_id)]
+
+
+def _results_to_docs_and_scores(
+    results: Any, include_id: Optional[bool] = False
+) -> List[Tuple[Document, float]]:
+    # TODO: Chroma can do batch querying,
+    # we shouldn't hard code to the 1st result
+    output = []
+
+    for result in zip(
+        results["documents"][0],
+        results["metadatas"][0],
+        results["distances"][0],
+        results["ids"][0],
+    ):
+        metadata = result[1] or {}
+        if include_id:
+            metadata["id"] = result[3]
```

Comment:
search logic should not be modifying metadata.

It's OK if it's present during indexing, but shouldn't be mutated on the search path, as the vectorstore should be returning the document as it was indexed.

Reply:
Yeah, this was an iffy change. I didnt want to create an entirely new key to store ID so went with just adding it to the metadata. But if we go with your suggestion from the first comment and setup a new ID attribute for documents, that would resolve such issues.